apps/managedidentity/azure_ml.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package managedidentity
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+)
+
+func createAzureMLAuthRequest(ctx context.Context, id ID, resource string) (*http.Request, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, os.Getenv(msiEndpointEnvVar), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("secret", os.Getenv(msiSecretEnvVar))
+	q := req.URL.Query()
+	q.Set(apiVersionQueryParameterName, azureMLAPIVersion)
+	q.Set(resourceQueryParameterName, resource)
+
+	switch t := id.(type) {
+	case UserAssignedClientID:
+		q.Set("clientid", string(t))
+	case systemAssignedValue:
+	default:
+		return nil, fmt.Errorf("unsupported type %T", id)
+	}
```

Comment:
Sorry I forgot to point this out before now, but `clientid` must always be set, and the platform provides a default value. So instead of the switch, you need something like this:
```go
q.Set("clientid", os.Getenv("DEFAULT_IDENTITY_CLIENT_ID"))
if cid, ok := id.(UserAssignedClientID); ok {
	q.Set("clientid", string(cid))
}
```